core/graph/src/main/java/uk/gov/gchq/gaffer/graph/hook/NamedViewResolver.java

```diff
@@ -0,0 +1,110 @@
+        cache = new NamedViewCache();
+    }
+
+    public NamedViewResolver(final NamedViewCache cache) {
+        this.cache = cache;
+    }
+
+    @Override
+    public void preExecute(final OperationChain<?> opChain, final Context context) {
+        resolveViewsInOperations(opChain);
+    }
+
+    @Override
+    public <T> T postExecute(final T result, final OperationChain<?> opChain, final Context context) {
+        return result;
+    }
+
+    @Override
+    public <T> T onFailure(final T result, final OperationChain<?> opChain, final Context context, final Exception e) {
+        return result;
+    }
+
+    private void resolveViewsInOperations(final Operations<?> operations) {
+        final List<Operation> updatedOperations = new ArrayList<>(operations.getOperations().size());
+
+        for (final Operation operation : operations.getOperations()) {
+            if (operation instanceof OperationView) {
+                if (((OperationView) operation).getView() instanceof NamedView) {
+                    final View resolvedView = resolveViewInOperation((NamedView) ((OperationView) operation).getView());
+                    ((NamedView) resolvedView).setName(null);
+                    ((NamedView) ((OperationView) operation).getView()).setName(null);
+                    final View viewMergedWithOriginalView = new View.Builder()
+                            .merge(resolvedView)
+                            .merge(((OperationView) operation).getView())
+                            .build();
+
+                    ((OperationView) operation).setView(viewMergedWithOriginalView);
+                }
+            } else {
+                if (operation instanceof Operations) {
+                    resolveViewsInOperations((Operations<?>) operation);
+                }
+            }
+            updatedOperations.add(operation);
+        }
+        operations.updateOperations((List) updatedOperations);
+    }
+
+    private NamedView resolveViewInOperation(final NamedView namedView) {
+        NamedView.Builder newNamedView;
+        try {
+            NamedViewDetail cachedNamedView = cache.getNamedView(namedView.getName());
+            View resolvedCachedView = cachedNamedView.getView(namedView.getParameters());
+            newNamedView = new NamedView.Builder()
+                    .name(namedView.getName())
+                    .merge(resolvedCachedView);
+
+            if (resolvedCachedView instanceof NamedView && null != ((NamedView) resolvedCachedView).getMergedNamedViewNames()) {
+                for (final String name : ((NamedView) resolvedCachedView).getMergedNamedViewNames()) {
+                    final NamedViewDetail nestedCachedNamedView = cache.getNamedView(name);
+                    final NamedView resolvedNestedCacheNamedView = new NamedView.Builder().name(null).merge(nestedCachedNamedView.getView(namedView.getParameters())).build();
+                    newNamedView.merge(resolvedNestedCacheNamedView);
```

Comment:
I think you need another layer of recursion to check that the resolved named view doesn't need further resolving.

Reply:
This won't be necessary, as NamedViews are merged all the lists are concatenated so the top level NamedView list of mergedNamedViewNames will contain all child NamedView names.  Tests have been added to check expected behaviour is correct.